test/distributed/_tensor/test_attention.py

```diff
@@ -299,18 +307,29 @@ def test_ring_attention_custom_transformer(self) -> None:
 
     @skip_if_lt_x_gpu(2)
     @unittest.skipIf(
-        not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Does not support flash attention"
+        not PLATFORM_SUPPORTS_FUSED_ATTENTION,
+        "Does not support flash nor efficient attention",
     )
+    @unittest.skipIf(
+        TEST_CUDA and not TEST_WITH_ROCM and not PLATFORM_SUPPORTS_FLASH_ATTENTION,
+        "Does not support flash attention",
+    )  # On CUDA (not ROCM) platform, the UT is skipped if no FA support (even if ME may get supported)
     @with_comms
     @parametrize(
         "attention_fn",
         [
-            _scaled_dot_product_ring_flash_attention,
-            _scaled_dot_product_ring_efficient_attention,
+            _scaled_dot_product_ring_flash_attention
+            if PLATFORM_SUPPORTS_FLASH_ATTENTION
+            else None,
+            _scaled_dot_product_ring_efficient_attention
+            if PLATFORM_SUPPORTS_MEM_EFF_ATTENTION
+            else None,
             # _scaled_dot_product_ring_cudnn_attention, # TODO: not built by default
         ],
     )
     def test_ring_attention_compile(self, attention_fn: object) -> None:
+        if attention_fn is None:
```

Comment:
is this another, "parametrize cant be empty" problem again?

Reply:
Passing `None` as `attention_fn` is a trick to ensure on every platform, `test_ring_attention_compile_attention_fn0` always refers to the test of FA, and `test_ring_attention_compile_attention_fn0` is always for ME.

(However I do agree the skipping message can be improved, but haven't found an elegant way)